### PR TITLE
Root to leaf path with given sum

### LINKED ISSUE #917: Root to leaf path with given sum in Binary tree
Would like to add an algorithm to check whether there is root to leaf path with its sum as S in C++.
Please assign this to me under gssoc2021.

### PR DESCRIPTION
**Pull request by tk315**
Kindly check it out and do the needful.
Thanks!

Fixed #917

### CODE REVIEW

**Review by tarun26091999 — CHANGES_REQUESTED** [review]
change 'Root to leaf path Sum' to 'Root to leaf path with given sum'

**Comment by tk315** [comment]
Requested Changes done

File: Code/C++/Root_to_leaf_path_with_given_sum.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+int main() {
+
+    int data;               
+    cin >> data;
+
+    Node* root = new Node(data);
+
+    while(true){ 
+        cin >> data;
+        if(data==-1)break;
```
Comment (raksha009) [inline Code/C++/Root_to_leaf_path_with_given_sum.cpp:57]:
```suggestion
        if(data == -1)break;
```

File: Code/C++/Root_to_leaf_path_with_given_sum.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    int sum;
+    cin >> sum;
+    cout << has_path_sum(root, sum) << endl;
+
+    return 0;
+}
+
+/*
+Input:
```
Comment (raksha009) [inline Code/C++/Root_to_leaf_path_with_given_sum.cpp:69]: Follow 

Test Cases

1.
Input: 
Output:

File: Code/C++/Root_to_leaf_path_with_given_sum.cpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    1
```
Comment (raksha009) [inline Code/C++/Root_to_leaf_path_with_given_sum.cpp:70]: In Input, add exactly the same way you enter from keyboard.

You may add this tree later in explanation part

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Code/C++/Root_to_leaf_path_with_given_sum.cpp (hunk 1 above)
Comment (tk315) [reply Code/C++/Root_to_leaf_path_with_given_sum.cpp]: done

File: Code/C++/Root_to_leaf_path_with_given_sum.cpp (hunk 2 above)
Comment (tk315) [reply Code/C++/Root_to_leaf_path_with_given_sum.cpp:73]: done

File: Code/C++/Root_to_leaf_path_with_given_sum.cpp (hunk 3 above)
Comment (tk315) [reply Code/C++/Root_to_leaf_path_with_given_sum.cpp:80]: done

File: Code/C++/Root_to_leaf_path_with_given_sum.cpp [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    cout << has_path_sum(root, sum) << endl;
+
+    return 0;
+}
+
+/*
+Test Cases:
+
+1.
+
```
Comment (raksha009) [inline Code/C++/Root_to_leaf_path_with_given_sum.cpp:72]: Remove extra lines 72 and 87.

File: Code/C++/Root_to_leaf_path_with_given_sum.cpp [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+Output:
+0
+
+Explanation:
+    1         
+     \  
+      7       
+     / \
+    4   9
```
Comment (raksha009) [inline Code/C++/Root_to_leaf_path_with_given_sum.cpp:84]: In explanation you would have to have to add which path you took too :)

File: Tree/readme.md [hunk 6]
```diff
@@ -52,7 +52,6 @@ that, this property should be satisfied at every node in the tree.
 * Construction of binary tree from preorder and inorder traversal ----> [C++](/Code/C++/binary_tree_from_preorder_and_inorder.cpp) 
 * Level Order Traversal in BST ----> [C++]() | [Java]() | [Python](/Code/Python/level_order_traversal_binary_tree.py)
 * Spiral Traversal of Binary Tree ----> [C++](/Code/C++/spiral_traversal_of_binary_tree.cpp) 
-
-
+* Root to leaf path with given sum ----> [C++](/Code/C++/Root_to_leaf_path_with_given_sum.cpp)
```
Comment (raksha009) [inline Tree/readme.md:55]:
```suggestion
* Root to Leaf Path with Given Sum ----> [C++](/Code/C++/Root_to_leaf_path_with_given_sum.cpp)
```

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Code/C++/Root_to_leaf_path_with_given_sum.cpp (hunk 5 above)
Comment (tk315) [reply Code/C++/Root_to_leaf_path_with_given_sum.cpp:83]: done

File: Code/C++/Root_to_leaf_path_with_given_sum.cpp (hunk 4 above)
Comment (tk315) [reply Code/C++/Root_to_leaf_path_with_given_sum.cpp]: done

File: Code/C++/Root_to_leaf_path_with_given_sum.cpp [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    
+2.
+Input:
+5 1 7 6 8 -1 6  
+
+Output:
+1
+
+Explanation:
+    / 5
```
Comment (raksha009) [inline Code/C++/Root_to_leaf_path_with_given_sum.cpp:94]: Hey, you seen to add / instead of space here.

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Review by raksha009 — APPROVED** [review]

**Review by tarun26091999 — CHANGES_REQUESTED** [review]
in the examples given, write the input in 3 lines, like this
4
2 5 4 7
12

otherwise it's very confusing.

**Comment by tk315** [comment]
@tarun26091999 Sir done

File: Code/C++/Root_to_leaf_path_with_given_sum.cpp [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+int main() {
+
+    int n;               
+    cin >> n;
+    
+    int data;
+    cin >> data;
+    Node* root = new Node(data);
+
+    for(int i=1;i<n;i++){
```
Comment (tarun26091999) [inline Code/C++/Root_to_leaf_path_with_given_sum.cpp:57]: ???

Comment (tk315) [reply Code/C++/Root_to_leaf_path_with_given_sum.cpp:57]: Sir to take the input in this style I need to do this.

Comment (tk315) [reply Code/C++/Root_to_leaf_path_with_given_sum.cpp:57]: earlier i was taking input untill the occurance of -1

**Review by tarun26091999 — APPROVED** [review]

**Pull request merged**